src/extensions/scratch3_boost/index.js

```diff
@@ -1788,19 +1788,23 @@ class Scratch3BoostBlocks {
         this._forEachMotor(args.MOTOR_ID, motorIndex => {
             const motor = this._peripheral.motor(motorIndex);
             if (motor) {
-                motor.power = MathUtil.clamp(Cast.toNumber(args.POWER), 0, 100);
-                switch (motor.status) {
-                case BoostMotorState.ON_FOREVER:
-                    motor.turnOnForever(false);
-                    break;
-                case BoostMotorState.ON_FOR_TIME:
-                    motor.turnOnFor(motor.pendingTimeoutStartTime + motor.pendingTimeoutDelay - Date.now(), false);
-                    break;
-                case BoostMotorState.ON_FOR_ROTATION: {
-                    const p = Math.abs(motor.pendingPositionDestination - motor.position, false);
-                    motor.turnOnForDegrees(p, Math.sign(p));
-                    break;
-                }
+                const power = MathUtil.clamp(Cast.toNumber(args.POWER), 0, 100);
+                // if motor power is already equal to the scaled input power, don't do anything
+                if (motor.power !== MathUtil.scale(power, 1, 100, 10, 100)) {
```

Comment:
`MathUtil.scale(power, 1, 100, 10, 100)` is what the setter for power for the motor uses.  If this overall fix is what we want to use to fix the bug in #2125, then we could maybe move that scaling to a helper function.